system/libarc32_arduino101/common/scss_registers.h

```diff
@@ -38,6 +38,8 @@
 #define MMIO_REG_ADDR(addr) ((volatile uint32_t *)(addr))
 #define MMIO_REG_VAL_FROM_BASE(base, offset) \
     (*((volatile uint32_t *)((base)+(offset))))
+    
+#define REG_VAL(addr) (*(uint32_t *)(addr))
```

Comment:
No need to make anything volatile-- MMIO_REG_VAL casts the passed pointer as a volatile, which forces volatile semantics on whatever that pointer happens to point to.

And, as was reinforced last week during our course-- you always use volatile to access memory mapped HW! So always using MMIO_REG_VAL is a good idea.